cpp/locality/AABBQuery.h

```diff
@@ -115,31 +118,31 @@ class AABBIterator : public NeighborQueryPerPointIterator
 {
 public:
     //! Constructor
-    AABBIterator(const AABBQuery* neighbor_query, const vec3<float> query_point, unsigned int query_point_idx,
-                 float r_max, float r_min, bool exclude_ii)
+    AABBIterator(const AABBQuery* neighbor_query, const vec3<float>& query_point,
+                 unsigned int query_point_idx, float r_max, float r_min, bool exclude_ii)
         : NeighborQueryPerPointIterator(neighbor_query, query_point, query_point_idx, r_max, r_min,
                                         exclude_ii),
           m_aabb_query(neighbor_query)
     {}
 
     //! Empty Destructor
-    virtual ~AABBIterator() {}
+    ~AABBIterator() override = default;
 
     //! Computes the image vectors to query for
     void updateImageVectors(float r_max, bool _check_r_max = true);
 
 protected:
     const AABBQuery* m_aabb_query;         //!< Link to the AABBQuery object
     std::vector<vec3<float>> m_image_list; //!< List of translation vectors
-    unsigned int m_n_images;               //!< The number of image vectors to check
+    unsigned int m_n_images {0};           //!< The number of image vectors to check
```

Comment:
Does this actually have any different behavior than what was there before? Initializing instance variables like this doesn't seem like good code convention to me.

Reply:
Yeah actually of all the changes that clang-tidy and cppcheck recommended, this syntax was the only thing that was completely new to me. It's documented [here](https://en.cppreference.com/w/cpp/language/data_members) (in the member initialization section).

I was at first hesitant to make this change, but I was convinced because it does have significant benefits over just using initializer lists in the constructor. This initializer becomes the default for all constructors, but can be overridden by constructor initializer lists. The big benefit here is that it prevents any issues where a constructor _forgets_ to initialize parameters, something that was the case in a few places in freud. The ancillary benefit is that it's nice and readable here, the default value is right next to the declaration (and hopefully, the docstring).

Reply:
It also seems to help limit the size of initializer lists, which can help with readability. Let's stick with it.